gallery/src/PageHeader/HostedVersionsSelect.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useEffect, ChangeEventHandler, useState } from 'react';
+import { NxFormSelect } from '@sonatype/react-shared-components';
+
+import packageJson from '../../package.json';
+
+import './HostedVersionsSelect.scss';
+
+const CURRENT_VERSION = packageJson.version;
+const HOSTED_VERSIONS_JSON_URL = 'https://gallery.sonatype.dev/hosted-versions.json';
+
+const getHostedVersions = () => fetch(HOSTED_VERSIONS_JSON_URL).then((response) => {
+  if (!response.ok) {
+    throw new Error('Failed to load hosted versions.');
+  }
+  return response.json();
+});
+
+const HostedVersionsSelect = () => {
+  const [isLoadingHostedVersions, setIsLoadingHostedVersions] = useState(false);
+  const [hostedVersions, setHostedVersions] = useState([CURRENT_VERSION]);
+
+  const onChange: ChangeEventHandler<HTMLSelectElement> = event =>
+    window.location.href = `/versions/${event.target.value}`;
+
+  useEffect(() => {
+    setIsLoadingHostedVersions(true);
+
+    const loadHostedVersions = async () => {
```

Comment:
Why is `loadHostedVersions` a separate function rather than having this logic inline?

Reply:
> Why is `loadHostedVersions` a separate function rather than having this logic inline?

This is to avoid useEffect complaining: `'() => Promise<void>' is not assignable to parameter of type 'EffectCallback'`